docs/reference/esql/time-spans.asciidoc

```diff
@@ -0,0 +1,72 @@
+[[esql-time-spans]]
+=== {esql} time spans
+
+++++
+<titleabbrev>Time spans</titleabbrev>
+++++
+
+Time spans provide a way of describing a specific period of time that separates two datetime values. There are currently two supported types of time spans, `DATE_PERIOD` specifies intervals in years, quarters, months, weeks and days, and `TIME_DURATION` specifies intervals in hours, minutes, seconds and milliseconds.
+
+The time span contains two elements, an integer value and a temporal unit, both of them are required when specifying a time span.
+
+Time spans are often used in datetime related grouping functions like <<esql-bucket, BUCKET>>, scalar functions like <<esql-date_trunc, DATE_TRUNC>> and arithmetic operators like `+` and `-`. There are conversions functions that can convert string literals to time spans, <<esql-to_dateperiod, TO_DATEPERIOD>>, <<esql-to_timeduration, TO_TIMEDURATION>> and their corresponding cast operators `::DATE_PERIOD` and `::TIME_DURATION`.
+
+[float]
+=== Examples of using time spans in {esql}
+
+With `BUCKET`
+[source, esql]
+----
+FROM employees
+| STATS hires_per_week = COUNT(*) BY week = BUCKET(hire_date, 1 week)
+| SORT week
+----
+
+With `DATE_TRUNC`
+[source, esql]
+----
+FROM employees
+| EVAL x = date_trunc(240 hours, hire_date)
+----
+
+With `+` and/or `-`
+[source, esql]
+----
+FROM employees
+| EVAL x = hire_date + 1 month - 1 day
+----
```

Comment:
Nit: it would be great if we could extract these from existing suites (dedicated or `docs.csv-spec`), or add these as new, to ensure they'll work across possible changes.

Reply:
Thanks for reminding! I'll change the examples to existing tests to make sure they work.